solver/result/attestation.go

```diff
@@ -4,44 +4,35 @@ import (
 	digest "github.com/opencontainers/go-digest"
 )
 
-type Attestation interface {
-	isAttestation()
-}
+const (
+	AttestationKindInToto = iota
+)
 
-type InTotoAttestation struct {
-	PredicateType   string
-	PredicateRefKey string
-	PredicatePath   string
-	Subjects        []InTotoSubject
-}
+const (
+	InTotoSubjectKindRaw = iota
+	InTotoSubjectKindSelf
+)
 
-func (a *InTotoAttestation) isAttestation() {}
+type Attestation struct {
+	Kind int
 
-type InTotoSubject interface {
-	isInTotoSubject()
+	Ref                 string
+	Path                string
+	InTotoPredicateType string
```

Comment:
Grouping intoto (kind specific) fields to a struct would add too much verbosity?